drv/stm32xx-i2c-server/src/main.rs

```diff
@@ -46,134 +46,207 @@ fn find_mux(
     controller: &I2cController<'_>,
     port: PortIndex,
     muxes: &[I2cMux<'_>],
-    mux: Option<(Mux, Segment)>,
-    mut func: impl FnMut(&I2cMux<'_>, Mux, Segment) -> Result<(), ResponseCode>,
+    id: Mux,
+    mut func: impl FnMut(&I2cMux<'_>) -> Result<(), ResponseCode>,
 ) -> Result<(), ResponseCode> {
-    match mux {
-        Some((id, segment)) => {
-            for mux in muxes {
-                if mux.controller != controller.controller {
-                    continue;
-                }
+    for mux in muxes {
+        if mux.controller != controller.controller {
```

Comment:
Nit: it's weird that the conditionals are broken up here.  We could instead write the body of the loop as a single (positive) conditional:
```rust
    for mux in muxes {
        if mux.controller == controller.controller
            && mux.port == port
            && mux.id == id
        {
            return func(mux);
        }
    }

    Err(ResponseCode::MuxNotFound)
```

Reply:
Yeah, I think that's broadly historical -- and agreed that phrasing it without the `continue`s is clearer; thank you!